File: java/sprykerFileCreator/model/definition/spryker/ZedControllerDefinition.java
package sprykerFileCreator.model.definition.spryker;

import pav.sprykerFileCreator.model.definition.ClassDefinitionInterface;
import pav.sprykerFileCreator.model.generator.SprykerConstants;

public class ZedControllerDefinition implements ClassDefinitionInterface {

    @Override
    public String getClassType() {
        return SprykerConstants.ZED_CONTROLLER;
    }

    @Override
    public String getNamePattern() {
        return "{{ className }}Controller";
    }

    @Override
    public String getNamespacePattern() {
        return "{{ projectName }}\\Zed\\{{ bundleName }}\\Communication\\Controller";
    }

    @Override
    public String getMethodForReturnType() {
        return "";
    }

    @Override
    public String[] getDocBlockClasses() {
        String [] docBlockClasses = {
                SprykerConstants.ZED_COMMUNICATION_FACTORY,
                SprykerConstants.ZED_QUERY_CONTAINER,
                SprykerConstants.ZED_FACADE
        };

        return docBlockClasses;
    }

    @Override
    public String getDefaultParentClass() {
        return "Spryker\\Zed\\Application\\Communication\\Controller\\AbstractController";
    }
}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/definitions/spryker/ZedControllerDefinition.kt
package com.github.tobi812.sprykerplugin.models.definitions.spryker

import com.github.tobi812.sprykerplugin.constants.SprykerConstants
import com.github.tobi812.sprykerplugin.models.definitions.ClassDefinitionInterface

class ZedControllerDefinition : ClassDefinitionInterface {
    override val classType: String
        get() = SprykerConstants.ZED_CONTROLLER
    override val namePattern: String
        get() = "{{ className }}Controller"
    override val namespacePattern: String
        get() = "{{ projectName }}\\Zed\\{{ bundleName }}\\Communication\\Controller"
    override val methodForReturnType: String
        get() = ""
    override val docBlockClasses: Array<String>
        get() = arrayOf(
                SprykerConstants.ZED_COMMUNICATION_FACTORY,
                SprykerConstants.ZED_QUERY_CONTAINER,
                SprykerConstants.ZED_FACADE
        )
    override val defaultParentClass: String
        get() = "Spryker\\Zed\\Application\\Communication\\Controller\\AbstractController"
}